File: plugins/types/OcrPlugin.java
package com.mrcomic.plugins.types;

import android.graphics.Bitmap;
import com.mrcomic.plugins.PluginInput;
import com.mrcomic.plugins.PluginResult;

/**
 * Интерфейс для OCR плагинов
 * Определяет методы для оптического распознавания текста на изображениях
 */
public interface OcrPlugin {
    
    /**
     * Распознать текст на изображении
     * 
     * @param image Изображение для распознавания
     * @param language Код языка для распознавания (ISO 639-1)
     * @return Результат распознавания текста
     */
    PluginResult recognizeText(Bitmap image, String language);
    
    /**
     * Получить список поддерживаемых языков
     * 
     * @return Массив кодов поддерживаемых языков (ISO 639-1)
     */
    String[] getSupportedLanguages();
    
    /**
     * Проверить, поддерживается ли указанный язык
     * 
     * @param language Код языка (ISO 639-1)
     * @return true, если язык поддерживается, иначе false
     */
    boolean isLanguageSupported(String language);
    
    /**
     * Получить точность распознавания
     * 
     * @return Точность распознавания в процентах (0-100)
     */
    int getAccuracy();
    
    /**
     * Установить точность распознавания
     * Более высокая точность может требовать больше ресурсов
     * 
     * @param accuracy Точность распознавания в процентах (0-100)
     */
    void setAccuracy(int accuracy);
}

File: app/src/main/java/com/mrcomic/plugins/ocr/OcrPlugin.kt
package com.mrcomic.plugins.ocr

import android.graphics.Bitmap

/**
 * Интерфейс для OCR плагинов
 */
interface OcrPlugin {
    
    /**
     * Инициализация плагина
     */
    suspend fun initialize(): Boolean
    
    /**
     * Распознавание текста на изображении
     */
    suspend fun recognizeText(bitmap: Bitmap, language: String = "auto"): OcrResult
    
    /**
     * Получение поддерживаемых языков
     */
    fun getSupportedLanguages(): List<String>
    
    /**
     * Освобождение ресурсов
     */
    fun cleanup()
}

/**
 * Результат OCR распознавания
 */
data class OcrResult(
    val text: String,
    val confidence: Float,
    val language: String,
    val textBlocks: List<TextBlockResult> = emptyList(),
    val processingTime: Long = 0L,
    val error: String? = null
)

/**
 * Результат распознавания текстового блока
 */
data class TextBlockResult(
    val text: String,
    val confidence: Float,
    val boundingBox: BoundingBox,
    val words: List<WordResult> = emptyList()
)

/**
 * Результат распознавания слова
 */
data class WordResult(
    val text: String,
    val confidence: Float,
    val boundingBox: BoundingBox
)

/**
 * Ограничивающий прямоугольник
 */
data class BoundingBox(
    val left: Int,
    val top: Int,
    val right: Int,
    val bottom: Int
) {
    val width: Int get() = right - left
    val height: Int get() = bottom - top
}

